### PR TITLE
React 18 upgrade

### PR DESCRIPTION
**Pull request by alonshmiel**
1. Update packages
2. Change React render
3. Fix State error because of the upgrade
4. Fix failed tests
5. Fix lint
6. Rename invalid class to rg-invalid (fixed some texts)

### CODE REVIEW

File: cypress/integration/common/utils.ts [hunk 1]
```diff
@@ -192,13 +192,15 @@ export class Utilities {
...
 
   click(x: number, y: number): void {
-    this.getScrollableElement().trigger("pointerdown", x, y, {
-      pointerType: "mouse",
-    });
-    this.getBody().trigger("pointerup", 0, 0, {
-      pointerType: "mouse",
-      force: true,
-    }); //
+    // TODO: use pointerdown + pointerup instead of click
```
Comment (fellmann) [inline cypress/integration/common/utils.ts:195]: Why was this changed? Short comment would be helpful.

File: package.json [hunk 2]
```diff
@@ -37,27 +37,26 @@
...
+    "rollup-plugin-scss": "^3.0.0",
     "rollup-plugin-terser": "^7.0.2",
-    "rollup-plugin-typescript2": "^0.30.0",
+    "rollup-plugin-typescript2": "^0.31.0",
     "typescript": "^4.2.3"
   },
   "peerDependencies": {
-    "react": "^16.13.1 || ^17.0.0",
-    "react-dom": "^16.13.1 || ^17.0.0"
+    "react": "^18.2.0",
```
Comment (fellmann) [inline package.json]: Do I understand correctly that the new version is only compatible with React 18? Or is React 17 still supported?

**Review by fellmann — COMMENTED** [review]
For me, quite some tests fail locally. For example, 7 failed tests in cellEditor.spec.ts. Do these tests work for you?

**Comment by alonshmiel** [comment]
Regarding to the failed tests,
In the main branch of Silevis, there are 18 tests that fail (If I remember the exact number).
In my branch, 8 tests fail (7 tests are common tests with main branch and the 8th test passes manually).

File: cypress/integration/common/utils.ts (hunk 1 above)
Comment (alonshmiel) [reply cypress/integration/common/utils.ts:195]: I think that it's related to cypress:
https://github.com/cypress-io/cypress/issues/21381

I didn't see that we import cypress/react so I did nothing about it but I will add a comment.

**Review by fellmann — APPROVED** [review]
